src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -0,0 +1,160 @@
+  private final SwerveModule backRightModule;
+
+  private ChassisSpeeds chassisSpeeds = new ChassisSpeeds(0.0, 0.0, 0.0);
+
+  public DrivetrainSubsystem() {
+    ShuffleboardTab tab = Shuffleboard.getTab("Drivetrain");
+
+    frontLeftModule = Mk4SwerveModuleHelper.createFalcon500(
+            // This parameter is optional, but will allow you to see the current state of the module on the dashboard.
+            tab.getLayout("Front Left Module", BuiltInLayouts.kList)
+                    .withSize(2, 4)
+                    .withPosition(0, 0),
+            // This can either be STANDARD or FAST depending on your gear configuration
+            Mk4SwerveModuleHelper.GearRatio.L1,// FIXME find the actual gear ratio
+            // This is the ID of the drive motor
+            FL_DRIVE_MOTOR_ID,
+            // This is the ID of the steer motor
+            FL_STEER_MOTOR_ID,
+            // This is the ID of the steer encoder
+            FL_STEER_ENCODER_ID,
+            // This is how much the steer encoder is offset from true zero (In our case, zero is facing straight forward)
+            FL_STEER_OFFSET
+    );
+
+    // We will do the same for the other modules
+    frontRightModule = Mk4SwerveModuleHelper.createFalcon500(
+            tab.getLayout("Front Right Module", BuiltInLayouts.kList)
+                    .withSize(2, 4)
+                    .withPosition(2, 0),
+            Mk4SwerveModuleHelper.GearRatio.L1,// FIXME find the actual gear ratio
+            FR_DRIVE_MOTOR_ID,
+            FR_STEER_MOTOR_ID,
+            FR_STEER_ENCODER_ID,
+            FR_STEER_OFFSET
+    );
+
+    backLeftModule = Mk4SwerveModuleHelper.createFalcon500(
+            tab.getLayout("Back Left Module", BuiltInLayouts.kList)
+                    .withSize(2, 4)
+                    .withPosition(4, 0),
+            Mk4SwerveModuleHelper.GearRatio.L1,// FIXME find the actual gear ratio
+            BL_DRIVE_MOTOR_ID,
+            BL_STEER_MOTOR_ID,
+            BL_STEER_ENCODER_ID,
+            BL_STEER_OFFSET
+    );
+
+    backRightModule = Mk4SwerveModuleHelper.createFalcon500(
+            tab.getLayout("Back Right Module", BuiltInLayouts.kList)
+                    .withSize(2, 4)
+                    .withPosition(6, 0),
+            Mk4SwerveModuleHelper.GearRatio.L1,// FIXME find the actual gear ratio
+            BR_DRIVE_MOTOR_ID,
+            BR_STEER_MOTOR_ID,
+            BR_STEER_ENCODER_ID,
+            BR_STEER_OFFSET
+    );
+  }
+
+  /**
+   * Sets the gyroscope angle to zero. This can be used to set the direction the robot is currently facing to the
+   * 'forwards' direction.
+   */
+  public void zeroGyroscope() {
+    // FIXME Remove if you are using a Pigeon
+    pigeon.setFusedHeading(0.0);
+
+
+  }
+
+  public Rotation2d getGyroscopeRotation() {
+    return Rotation2d.fromDegrees(pigeon.getFusedHeading());
+  }
+
+  public void drive(ChassisSpeeds chassisSpeeds) {
+    chassisSpeeds = chassisSpeeds;
```

Comment:
@Dinodigger3000 Review this line... what would it be currently doing vs what should it be doing?